packages/hardhat/contracts/periphery/ControllerHelper.sol

```diff
@@ -189,6 +197,47 @@ contract ControllerHelper is FlashControllerHelper, IERC721Receiver {
         emit FlashWBurn(msg.sender, _vaultId, _wPowerPerpAmountToBurn, _collateralToWithdraw, _wPowerPerpAmountToBuy);
     }
 
+    /**
+     * @notice sell long wPowerPerp and flashswap mint short position
+     * @dev flahswap amount = collateral amount - msg.value - ETH from selling long wPowerPerp
+     * @param _vaultId vault ID
+     * @param _wPowerPerpAmountToMint wPowerPerp amount to mint
+     * @param _collateralAmount collateral amount to use for minting
+     * @param _wPowerPerpAmountToSell long wPowerPerp amount to sell
+     * @param _minToReceive min ETH amount to receive for selling long wPowerPerp amount
+     */
+    function flashswapSellLongWMint(
```

Comment:
this can be simplified/cheaper as 1 exactInFlashSwap right (similar to the sell long, mint + sell)

- do an exactInFlashSwap for the _wPowerPerpAmountToSell + _wPowerPerpAmountToMint (this requires slippage in terms of both quantities, so the last param is the minToReceive for the total sell)
- mint _wPowerPerpAmount using _collateralAmount (which can be a combo of msg.value and proceeds from step 1)
- repay #1 using the minted amount and the transferred in oSQTH

Reply:
looks good!